linkd/src/cfg/seed.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright © 2021 The Radicle Link Contributors
+//
+// This file is part of radicle-link, distributed under the GPLv3 with Radicle
+// Linking Exception. For full terms see the included LICENSE file.
+
+use std::{convert::TryFrom, io, net::SocketAddr};
+
+use librad::net::discovery;
+use tokio::net::lookup_host;
+
+use librad::PeerId;
+
+use crate::args::Bootstrap;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("the seed `{0}` failed to resolve to an address")]
+    DnsLookupFailed(String),
+
+    #[error(transparent)]
+    Io(#[from] io::Error),
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Seed {
+    pub addrs: Vec<SocketAddr>,
+    pub peer_id: PeerId,
+}
+
+impl Seed {
+    /// Create a [`Seed`] from a [`Bootstrap`].
+    ///
+    /// # Errors
+    ///
+    /// * If the supplied address cannot be resolved.
+    async fn from_bootstrap(bootstrap: &Bootstrap) -> Result<Self, Error> {
```

Comment:
```suggestion
    async fn try_from_bootstrap(bootstrap: &Bootstrap) -> Result<Self, Error> {
```

For idiotmaticness

Reply:
It would be more idiomatic if the advent of async wouldn't been incompatible with a majority of language idioms...

485bfbd

Reply:
That's why it's idiotmatic =]